dds_web/api/user.py

```diff
@@ -47,62 +48,103 @@ class AddUser(flask_restful.Resource):
+        existing_user = user_schemas.UserSchema().load({"email": email})
+        unanswered_invite = user_schemas.UnansweredInvite().load({"email": email})
 
         if existing_user or unanswered_invite:
-            if project and role:
-                whom = existing_user or unanswered_invite
-                add_user_result = self.add_to_project(whom, project, role)
-                flask.current_app.logger.debug(f"Add user result?: {add_user_result}")
-                return add_user_result, add_user_result["status"]
-            else:
+            if not project:
                 raise ddserr.DDSArgumentError(
-                    message="This user was already added to the system. Specify the project you wish to give access to."
+                    message=(
+                        "This user was already added to the system. "
+                        "Specify the project you wish to give access to."
+                    )
                 )
 
+            add_user_result = self.add_to_project(
+                whom=existing_user or unanswered_invite, project=project, role=role
+            )
+            return add_user_result, add_user_result["status"]
+
         else:
             # Send invite if the user doesn't exist
-            if project:
-                invite_user_result = self.invite_user(args, project)
-            else:
-                invite_user_result = self.invite_user(args)
+            invite_user_result = self.invite_user(email=email, new_user_role=role, project=project)
 
             return invite_user_result, invite_user_result["status"]
 
     @staticmethod
     @logging_bind_request
-    def invite_user(args, project=None):
+    def invite_user(email, new_user_role, project=None):
         """Invite a new user"""
 
-        try:
-            # Use schema to validate and check args, and create invite row
-            new_invite = user_schemas.InviteUserSchema().load(args)
+        current_user_role = get_user_roles_common(user=auth.current_user())
 
-        except ddserr.InviteError as invite_err:
+        if not project:
+            if current_user_role == "Project Owner":
+                raise ddserr.DDSArgumentError(
+                    message="Project ID required to invite users to projects."
+                )
+            if new_user_role == "Project Owner":
+                raise ddserr.DDSArgumentError(
+                    message="Project ID required to invite a 'Project Owner'."
+                )
+
+        # Verify role or current and new user
+        if current_user_role == "Super Admin" and project:
             return {
-                "message": invite_err.description,
                 "status": ddserr.InviteError.code.value,
+                "message": (
+                    "Super Admins do not have project data access and can therefore "
+                    "not invite users to specific projects."
+                ),
```

Comment:
Why is this error written different from the others? Any special reason?
For example, why not `raise InviteError(message="Super Admins do not have project data access and can therefore not invite users to specific projects.")`

Reply:
The raises are actually incorrect (or actually it's better to raise and then catch but this is another refactoring task), the `invite_user` function needs to be called multiple times from for exemple the CreateProject endpoint. And it shouldn't raise anything for a specific user getting invited.